### DISCUSSION TITLE
How do I create a set type in v?

### DISCUSSION THREAD

**Discussion by mark-summerfield** [vlang/v#10092] [General]
I want to create a set type that holds strings. At the moment I'm using `map[string]bool` but I only care about the key and don't need values. Is this possible in v? (I don't want to implement sets from scratch.) What I need is fast membership testing and fast add/remove; ordering doesn't matter.

**Comment by JalonSolov** [comment]
I don't know if it would be "fast" enough for you, but you could just create a string array and use the `in` operator to test for existence.

https://github.com/vlang/v/blob/master/doc/docs.md#in-operator

**Comment by mark-summerfield** [comment]
The set of words will have approx 4000 words in it and a linear search isn't good enough. I want a set or failing that a hash/dict/map.
